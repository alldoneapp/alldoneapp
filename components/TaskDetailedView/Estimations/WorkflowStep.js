import React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import styles, { colors } from '../../styles/global'
import UserTag from '../../Tags/UserTag'
import moment from 'moment'
import { useSelector } from 'react-redux'
import Button from '../../UIControls/Button'
import SharedHelper from '../../../utils/SharedHelper'
import { getDateFormat } from '../../UIComponents/FloatModals/DateFormatPickerModal'
import { translate } from '../../../i18n/TranslationService'
import { getEstimationIconByValue, getEstimationTagText } from '../../../utils/EstimationHelper'
import { getUserPresentationData } from '../../ContactsView/Utils/ContactsHelper'

export default function WorkflowStep({
    onPress,
    showModal,
    currentEstimation,
    isCurrentStep,
    stepNumber,
    step,
    projectId,
    disabled,
}) {
    const isMiddleScreen = useSelector(state => state.isMiddleScreen)
    const loggedUser = useSelector(state => state.loggedUser)
    const { description, reviewerUid, addedById, date, id } = step
    const reviewerData = getUserPresentationData(reviewerUid)
    const addedByData = getUserPresentationData(addedById)
    const info = isMiddleScreen
        ? translate('User added on Date', {
              user: addedByData.shortName,
              date: moment(date).format(getDateFormat()),
          })
        : translate('Step added by User, Date', {
              user: addedByData.displayName,
              date: moment(date).format(getDateFormat()),
          })
    const accessGranted = SharedHelper.accessGranted(loggedUser, projectId)

    return (
        <TouchableOpacity
            style={[localStyles.container, isCurrentStep ? { backgroundColor: colors.UtilityBlue100 } : undefined]}
            onPress={() => onPress(id)}
            disabled={disabled}
        >
            <View style={localStyles.stepNumberContainer}>
                <Text style={[styles.subtitle1, { color: 'white' }]}>{stepNumber}</Text>
            </View>

            <View style={localStyles.description}>
                <Text numberOfLines={1} style={[styles.body1, { color: colors.Text02 }]}>
                    {description}
                </Text>
                <Text numberOfLines={1} style={[styles.caption2, { color: colors.Text03 }]}>
                    {info}
                </Text>
            </View>

            <View style={localStyles.rightSection}>
                <View style={localStyles.sendToContainer}>
                    {!isMiddleScreen && (
                        <Text style={[styles.body2, { color: colors.Text02 }]}>{translate('Send to')}:</Text>
                    )}
                    <View style={!isMiddleScreen && { marginLeft: 12 }}>
                        <UserTag
                            user={{ displayName: reviewerData.displayName, photoURL: reviewerData.photoURL }}
                            onlyPhoto={isMiddleScreen}
                        />
                    </View>
                </View>

                <View style={{ marginLeft: 12 }}>
                    <Button
                        title={!isMiddleScreen && translate(getEstimationTagText(projectId, currentEstimation))}
                        type={'ghost'}
                        icon={`count-circle-${getEstimationIconByValue(projectId, currentEstimation)}`}
                        onPress={showModal}
                        disabled={!accessGranted || disabled}
                        buttonStyle={isMiddleScreen && { paddingRight: 0 }}
                    />
                </View>
            </View>
        </TouchableOpacity>
    )
}

const localStyles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        height: 64,
        alignItems: 'center',
        borderRadius: 4,
        borderWidth: 1,
        borderColor: colors.Grey300,
        paddingLeft: 5,
        paddingRight: 8,
        marginBottom: 16,
    },
    stepNumberContainer: {
        backgroundColor: colors.Gray400,
        borderRadius: 2,
        width: 28,
        height: 54,
        alignItems: 'center',
        justifyContent: 'center',
    },
    description: {
        flex: 1,
        flexDirection: 'column',
        marginLeft: 7,
    },
    sendToContainer: {
        flexDirection: 'row',
        height: 24,
        borderLeftWidth: 1,
        borderLeftColor: colors.Grey200,
        paddingLeft: 12,
    },
    rightSection: {
        marginLeft: 'auto',
        flexDirection: 'row',
        alignItems: 'center',
    },
})
