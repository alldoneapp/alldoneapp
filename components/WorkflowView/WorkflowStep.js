import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import styles, { colors } from '../styles/global'
import UserTag from '../Tags/UserTag'
import moment from 'moment'
import Spinner from '../UIComponents/Spinner'
import { useSelector } from 'react-redux'
import { getDateFormat } from '../UIComponents/FloatModals/DateFormatPickerModal'
import { translate } from '../../i18n/TranslationService'
import { getUserPresentationData } from '../ContactsView/Utils/ContactsHelper'

const WorkflowStep = ({ step, stepNumber, updatingStep }) => {
    const isMiddleScreen = useSelector(state => state.isMiddleScreen)
    const addedByData = getUserPresentationData(step.addedById)
    const reviewerData = getUserPresentationData(step.reviewerUid)
    const info = isMiddleScreen
        ? translate('User added on Date', {
              user: addedByData.shortName,
              date: moment(step.date).format(getDateFormat()),
          })
        : translate('Step added by User, Date', {
              user: addedByData.displayName,
              date: moment(step.date).format(getDateFormat()),
          })

    return (
        <View style={localStyles.container}>
            <View style={localStyles.stepNumberContainer}>
                <Text style={[styles.subtitle1, { color: 'white' }]}>{stepNumber}</Text>
            </View>

            <View style={localStyles.description}>
                <Text style={[styles.body1, { color: colors.Text01 }]}>{step.description}</Text>
                <Text style={[styles.caption2, { color: colors.Text03 }]}>{info}</Text>
            </View>

            <View style={localStyles.sendToContainer}>
                {!isMiddleScreen && (
                    <Text style={[styles.body2, { color: colors.Text02 }]}>{`${translate('Send to')}:`}</Text>
                )}
                <View style={!isMiddleScreen && { marginLeft: 12 }}>
                    {!updatingStep ? (
                        <UserTag
                            user={{
                                displayName: reviewerData.displayName,
                                photoURL: getUserPresentationData(step.reviewerUid).photoURL,
                            }}
                            onlyPhoto={isMiddleScreen}
                        />
                    ) : (
                        <Spinner containerSize={24} spinnerSize={16} />
                    )}
                </View>
            </View>
        </View>
    )
}

export default WorkflowStep

const localStyles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        height: 64,
        alignItems: 'center',
        borderRadius: 4,
        borderWidth: 1,
        borderColor: colors.UtilityBlue150,
        paddingLeft: 5,
        paddingRight: 8,
        marginBottom: 16,
    },
    stepNumberContainer: {
        backgroundColor: colors.UtilityBlue200,
        borderRadius: 2,
        width: 28,
        height: 54,
        alignItems: 'center',
        justifyContent: 'center',
    },
    description: {
        flexDirection: 'column',
        marginLeft: 7,
    },
    sendToContainer: {
        flexDirection: 'row',
        alignItems: 'center',
        marginLeft: 'auto',
        height: 24,
        borderLeftWidth: 1,
        borderLeftColor: colors.Grey200,
        paddingLeft: 12,
    },
})
