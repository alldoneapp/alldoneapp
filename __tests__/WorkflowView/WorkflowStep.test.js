import React from 'react'
import WorkflowStep from '../../components/WorkflowView/WorkflowStep'

import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import store from '../../redux/store'

describe('WorkflowStep component', () => {
    const step = { description: 'a', addedBy: 'b', date: Date.now(), reviewerName: 'c', photoURL: 'd' }
    describe('WorkflowStep snapshot test', () => {
        it('should render correctly', () => {
            const tree = renderer
                .create(
                    <Provider store={store}>
                        <WorkflowStep stepNumber={1} step={step} />
                    </Provider>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
