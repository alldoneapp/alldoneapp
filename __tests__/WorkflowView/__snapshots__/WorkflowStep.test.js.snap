// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowStep component WorkflowStep snapshot test should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#A3D1FF",
      "borderRadius": 4,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 64,
      "marginBottom": 16,
      "paddingLeft": 5,
      "paddingRight": 8,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#5AACFF",
        "borderRadius": 2,
        "height": 54,
        "justifyContent": "center",
        "width": 28,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Roboto-Medium",
            "fontSize": 16,
            "letterSpacing": 0.16,
            "lineHeight": 24,
          },
          Object {
            "color": "white",
          },
        ]
      }
    >
      1
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "marginLeft": 7,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#04142F",
            "fontFamily": "Roboto-Regular",
            "fontSize": 16,
            "letterSpacing": 0.32,
            "lineHeight": 24,
          },
          Object {
            "color": "#04142F",
          },
        ]
      }
    >
      a
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#04142F",
            "fontFamily": "Roboto-Regular",
            "fontSize": 12,
            "letterSpacing": 0.48,
            "lineHeight": 20,
          },
          Object {
            "color": "#8A94A6",
          },
        ]
      }
    >
      Step added by b, 22.04.2019
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderLeftColor": "#F1F3F4",
        "borderLeftWidth": 1,
        "flexDirection": "row",
        "height": 24,
        "marginLeft": "auto",
        "paddingLeft": 12,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#04142F",
            "fontFamily": "Roboto-Regular",
            "fontSize": 14,
            "letterSpacing": 0.32,
            "lineHeight": 22,
          },
          Object {
            "color": "#4E5D78",
          },
        ]
      }
    >
      Send to :
    </Text>
    <View
      style={
        Object {
          "marginLeft": 12,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#E7ECEF",
                  "borderRadius": 12,
                  "flexDirection": "row",
                  "height": 24,
                  "paddingLeft": 2,
                  "paddingRight": 10,
                },
                false,
              ]
            }
          >
            <Image
              source={
                Object {
                  "uri": undefined,
                }
              }
              style={
                Object {
                  "borderRadius": 100,
                  "height": 20,
                  "width": 20,
                }
              }
            />
            <View
              style={
                Object {
                  "paddingLeft": 4,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#04142F",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 14,
                      "letterSpacing": 0.16,
                      "lineHeight": 22,
                    },
                    Object {
                      "color": "#8A94A6",
                    },
                  ]
                }
              >
                c
              </Text>
            </View>
          </View>
        </Text>
      </View>
    </View>
  </View>
</View>
`;
